documents/api/filtersets.py

```diff
@@ -1,8 +1,48 @@
-from django_filters import rest_framework as filters
+from django.db.models import TextField
+from django.db.models.functions import Cast
+from django_filters import CharFilter, Filter, rest_framework as filters
+from django_filters.constants import EMPTY_VALUES
+from rest_framework.exceptions import ValidationError
+
+from utils import uuid
 
 from ..models import Document
 
 
+class UserUUIDFilter(CharFilter):
+    """
+    Allows filtering for null values in addition to valid UUID
+    """
+
+    def filter(self, qs, value):
+        if value in EMPTY_VALUES:
+            return qs
+        elif value.lower() == "null":
+            return qs.filter(user__isnull=True)
+        try:
+            value = uuid.UUID(value)
```

Comment:
Is this neccessary? As if UUID is not valid it should be impossible to find UUID which would match.

Reply:
I guess it's not necessary in that regard. I matched it with the implementation of Django's UUID filter.